tiatoolbox/annotation/storage.py

```diff
@@ -1592,13 +1767,100 @@ def _query(
             query_parameters["where"] = where
         # Predicate is a string
         if isinstance(where, str):
             sql_predicate = eval(where, SQL_GLOBALS, {})  # skipcq: PYL-W0123
             query_string += f" AND {sql_predicate}"
 
+        return query_string, query_parameters
+
+    def _query(
+        self,
+        columns: str,
+        geometry: Optional[Geometry] = None,
+        callable_columns: Optional[str] = None,
+        geometry_predicate="intersects",
+        where: Optional[Predicate] = None,
+        unique: bool = False,
+        no_constraints_ok: bool = False,
+        index_warning: bool = False,
+    ) -> sqlite3.Cursor:
+        """Common query construction logic for `query` and `iquery`.
+
+        Args:
+            columns(str):
+                The columns to select.
+            geometry(tuple or Geometry):
+                The geometry being queries against.
+            select_callable(str):
+                The rows to select when a callable is given to `where`.
+            callable_columns(str):
+                The columns to select when a callable is given to
+                `where`.
+            where (str or bytes or Callable):
+                The predicate to evaluate against candidate properties
+                during the query.
+            unique(bool):
+                Whether to return only unique results. Defaults to
+                False.
+            no_constraints_ok(bool):
+                Whether to allow the query to return results without
+                constraints (e.g. when the geometry or where predicate
+                is not provided). Defaults to False.
+            index_warning(bool):
+                Whether to warn if the query is not using an index.
+                Defaults to False.
+
+        Returns:
+            sqlite3.Cursor:
+                A database cursor for the current query.
+
+        """
+        if not no_constraints_ok and all(x is None for x in (geometry, where)):
+            raise ValueError("At least one of `geometry` or `where` must be specified.")
+        query_geometry = geometry
+        if callable_columns is None:
+            callable_columns = columns
+        if geometry_predicate not in self._geometry_predicate_names:
+            raise ValueError(
+                "Invalid geometry predicate."
+                f"Allowed values are: {', '.join(self._geometry_predicate_names)}."
+            )
+        cur = self.con.cursor()
+
+        # Normalise query geometry and determine if it is a rectangle
+        if isinstance(query_geometry, Iterable):
+            query_geometry = Polygon.from_bounds(*query_geometry)
+
+        if isinstance(where, Callable):
+            columns = callable_columns
+
+        query_parameters = {}
+
+        if isinstance(columns, bytes):
+            query_parameters["select"] = columns
+            columns = "pickle_expression(:select, properties) "
```

Comment:
This is not being covered by tests. As far as I can tell, its not actually reachable because a pickled select is handled elsewhere in the _handle_pickle_callable_pquery, and a pickled select can't ever be handed to _query.

I think we can remove, unless I have missed something

Reply:
pushed a commit to remove it